### PR TITLE
Zoma/simplify condition

### PR DESCRIPTION
**Pull request by coldplaying42**
Fixes #2653

## Proposed Changes
For condition and priority in .dialog files, they only accept expression. Therefore, we need to export the required functions. And rewrite the condition like  "condition": "lib.helpCondition($retries, $lastIntent)". All the global parameters are required to pass to the exported function.

There is an issue that when the kind is "Microsoft.OnEndOfActions", if the condition is "lib.missingCondition()", the missingCondition function could not be recognized.

### CODE REVIEW

File: experimental/generation/generator/packages/library/templates/generator.lg [hunk 1]
```diff
@@ -297,9 +297,22 @@ ${'#'} ${property}Entity_getValueText(val)
 {
     "$schema": "${appSchema}",
     "$kind": "Microsoft.OnEndOfActions",
-    "condition":"=!$${property} || $PropertyToChange == '${property}'",
-    "priority": "=indexOf(dialog.requiredProperties, '${property}')",
-    "actions": [       
+    "condition":"=!$${property} || $PropertyToChange == '${property}'", 
+    "priority": "=lib.missingPriority(dialog.requiredProperties, '${property}')",
```
Comment (chrimc62) [inline experimental/generation/generator/packages/library/templates/generator.lg:301]: >dialog.requiredProperties,  [](start = 38, length = 27)

Build reference into lib.missingPriority.

File: experimental/generation/generator/packages/library/templates/generator.lg [hunk 2]
```diff
@@ -297,9 +297,22 @@ ${'#'} ${property}Entity_getValueText(val)
...
+    "actions": [
+        {
+            "$kind": "Microsoft.IfCondition",
+            "condition": "lib.helpCondition($retries, $lastIntent)",
```
Comment (chrimc62) [inline experimental/generation/generator/packages/library/templates/generator.lg:305]: >$retries, $lastIntent [](start = 44, length = 21)

Build into lib.helpCondition.

File: experimental/generation/generator/packages/library/templates/generator.lg (hunk 1 above)
Comment (coldplaying42) [reply experimental/generation/generator/packages/library/templates/generator.lg:301]: do you mean that "priority" is kind of like lib.missingPriority('${property}')"  and the missingPriority function as follows:
# missingPriority(property)
- ${indexOf(dialog.requiredProperties, property)}

File: experimental/generation/generator/packages/library/templates/generator.lg [hunk 3]
```diff
@@ -297,9 +297,22 @@ ${'#'} ${property}Entity_getValueText(val)
...
-    "actions": [       
+    "condition":"=!$${property} || $PropertyToChange == '${property}'", 
+    "priority": "=lib.missingPriority('${property}')",
+    "actions": [
+        {
+            "$kind": "Microsoft.IfCondition",
+            "condition": "lib.helpCondition()",
+            "actions": [
+                {
+                    "$kind": "Microsoft.Ask",
```
Comment (chrimc62) [inline experimental/generation/generator/packages/library/templates/generator.lg:308]: >"Microsoft.Ask" [](start = 29, length = 15)

Should be Microsoft.SendActivity

File: experimental/generation/generator/packages/library/templates/generator.lg [hunk 4]
```diff
@@ -297,9 +297,22 @@ ${'#'} ${property}Entity_getValueText(val)
 {
     "$schema": "${appSchema}",
     "$kind": "Microsoft.OnEndOfActions",
-    "condition":"=!$${property} || $PropertyToChange == '${property}'",
-    "priority": "=indexOf(dialog.requiredProperties, '${property}')",
-    "actions": [       
+    "condition":"=!$${property} || $PropertyToChange == '${property}'",
```
Comment (chrimc62) [inline experimental/generation/generator/packages/library/templates/generator.lg:300]: >=!$${property} || $PropertyToChange == '${property}' [](start = 17, length = 52)

Should also be a lib function.

**Comment by mrivera-ms** [comment]
Hello @coldplaying42 , this PR was closed accidentally during the process of renaming the branch 'master' to 'main'. Please open a new PR that references the new 'main' branch. I apologize for the inconvenience.

**Pull request closed without merging**